### DISCUSSION TITLE
Portrait Orientation Only

### DISCUSSION THREAD

**Discussion by stanpanchenkosommoio** [buildnatively/community#33] [Ideas]
### Use case

```Text
a checkbox that when checked creates builds that are only in portrait orientation, not landscape
```

### Idea

```Text
Michael Scola
I'm going to +1 for Android here

·
September 9, 2023
·
Reply

member badge
Roman Furman
Hey, Sam! Tomorrow we will publish update, you will see this option in Appearance -> Style -> iOS Orientation

·
June 9, 2023
·
Reply
Sam
Roman Furman: Let's go! Thanks Roman 🙏

·
June 9, 2023
·
Reply

member badge
Roman Furman
Hey Sam! I think we have such functionality already implemented for iOS (we can expose it back if needed)
But not sure if Android will come soon
Will iOS only works for you?

·
June 5, 2023
·
Reply
Sam
Roman Furman: Hey! That's great. How do I add this screen orientation lock functionality to my iOS app?

·
June 5, 2023
·
Reply

member badge
Roman Furman
Sam: I've asked dev team to take a look on this, we will let you know once it's added. For now you can lock orientation by this actions: https://docs.buildnatively.com/guides/integration/control-style-and-colors#change-and-lock-device-orientation-in-the-app

1 like
·
June 6, 2023
·
Reply
Sam
Roman Furman: That’s perfect. Thank you very much!
```

**Comment by VaikoHansson** [comment]
+1 on Android.

We are building a game and as most games are played in landscape mode, we would like for our game to be in landscape also. It also helps with our UI.
We are also building another app that only works on portrait and if we don't handle this somehow, it will look very broken for the user.

**Reply by vdiaza** [reply]
maybe you can try using the JS screen orientation API

`screen.orientation.lock('landscape')`

**Comment by mahirf** [comment]
@romanfurman6 Can we get this going for Android? Facing the same issue and the current use case is an Android only build.

**Comment by romanfurman6** [comment]
Planning to deliver this by the end of this month

**Reply by mahirf** [reply]
Hey @romanfurman6 

Do you have an update on this? Going to rewrap an app now and this feature would be great to add before that! Can you share an expected timeline if it's not done?

**Reply by romanfurman6** [reply]
Hey @mahirf we're going to release a huge update somewhere between 15th and 19th July, please join our [Discord community](link.buildnatively.com/discord) to not miss it

**Reply by mahirf** [reply]
Thanks @romanfurman6. Looking forward to it!

**Reply by mahirf** [reply]
Hi @romanfurman6, 
Got an update on discord for a contest but nothing on the above. Do you recommend we go ahead and create a new build without the new features?

**Reply by romanfurman6** [reply]
> Hi @romanfurman6, 
> Got an update on discord for a contest but nothing on the above. Do you recommend we go ahead and create a new build without the new features?

I think should be live now, if no pls ping @victormagdalyuk

**Reply by mahirf** [reply]
The docs are not updated yet, but is this the setting for that? Under orientation?

![image](https://github.com/user-attachments/assets/83829cc7-5be2-462b-ad52-7629e0058d3e)

**Reply by victormagdalyuk** [reply]
Hi @mahirf ,
That's exactly the settings for Orientation on both Android and iOS.

**Reply by mahirf** [reply]
Amazing @victormagdalyuk. Thanks!
